### PR TITLE
[TACHYON-1432] Support flexible worker ports

### PR DESCRIPTION
**Pull request by aaudiber**
https://tachyon.atlassian.net/browse/TACHYON-1432

### CODE REVIEW

File: common/src/main/java/tachyon/util/network/NetworkAddressUtils.java [hunk 1]
```diff
@@ -167,17 +167,13 @@ public int getDefaultPort() {
...
 
-  public static void assertValidPort(final int port, TachyonConf tachyonConf) {
-    Preconditions.checkNotNull(tachyonConf);
+  public static void assertValidPort(final int port) {
     Preconditions.checkArgument(port < 65536, "Port must be less than 65536");
-
-    if (!tachyonConf.getBoolean(Constants.IN_TEST_MODE)) {
-      Preconditions.checkArgument(port > 0, "Port is only allowed to be zero in test mode.");
-    }
+    Preconditions.checkArgument(port >= 0, "Port must be non-negative");
```
Comment (calvinjia) [inline common/src/main/java/tachyon/util/network/NetworkAddressUtils.java]: Why do we allow port to be 0 in all cases now?

Comment (aaudiber) [inline common/src/main/java/tachyon/util/network/NetworkAddressUtils.java]: This allows us to have multiple workers on the same machine without their ports colliding. I'll add checks to make sure the master port isn't 0.

**Comment by aaudiber** [comment]
Jenkins, test this please.

**Comment by apc999** [comment]
+1 LGTM

**Comment by calvinjia** [comment]
LGTM

File: servers/src/main/java/tachyon/worker/netty/NettyDataServer.java [hunk 2]
```diff
@@ -118,7 +118,7 @@ public String getBindHost() {
   }
 
   /**
-   * Gets the port listening on.
+   * Gets the port the server is listening on.
```
Comment (jsimsa) [inline servers/src/main/java/tachyon/worker/netty/NettyDataServer.java]: do we need javadoc for an overriden method?

**Comment by jsimsa** [comment]
LGTM, just one minor comment

**Pull request merged**